File: communityWbSpringBoot/communityDB/src/main/java/dev/communitywb/communityWB/DataModel/HandBill.java
package dev.communitywb.communityWB.DataModel;



import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "handbill")
public class HandBill {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String s3_key;
    private String caption;
    private Double width;
    private Double height;
    private String address;

    @Embedded
    private LatLng location;
    

    @Override
    public String toString() {
        return "HandBill{" +
                "id=" + id +
                ", caption='" + caption + '\'' +
                ", width=" + width +
                ", height=" + height +
                ", address='" + address + '\'' +
                ", location=" + location +
                '}';
    }
}

File: communitywb/src/models/HandBill.ts
import { LatLng } from "leaflet";
import { HandBillMedia } from "./HandBillMedia";
import { User } from "./User";

export interface HandBill {
    id: number;
    caption: string;
    file: File;
    location: LatLng | null;
    address: string;
    width: number;
    height: number;
    user: User;
    handbillMedia: HandBillMedia[];
    positionX: number;
    positionY: number;
    createdAt: string;
    toFormData?(): FormData;
}